### ISSUE TITLE
Create test playbooks

### ISSUE THREAD

**Issue by mdellweg** [theforeman/foreman-ansible-modules#77]
- [x] foreman_architecture
- [x] foreman_auth_source_ldap
- [x] foreman_bookmark
- [x] foreman_compute_attribute
- [x] foreman_compute_profile
- [x] foreman_compute_resource
- [x] foreman_config_group
- [x] foreman_domain
- [x] foreman_environment
- [x] foreman_global_parameter
- [x] foreman_host
- [x] foreman_host_power (tested as part of foreman_host)
- [x] foreman_hostgroup
- [x] foreman_installation_medium
- [x] foreman_job_template
- [x] foreman_location
- [x] foreman_operating_system
- [x] foreman_organization
- [x] foreman_os_default_template
- [x] foreman_provisioning_template
- [x] foreman_ptable
- [x] foreman_realm
- [x] foreman_role
- [x] foreman_search_facts
- [x] foreman_setting
- [ ] foreman_setting_facts (deprecated)
- [x] foreman_snapshot
- [x] foreman_subnet
- [x] foreman_user
- [x] katello_activation_key
- [x] katello_content_credential
- [x] katello_content_view
- [x] katello_content_view_filter
- [ ] katello_content_view_publish (deprecated)
- [x] katello_content_view_version (tested as part of katello_content_view)
- [ ] katello_content_view_version_promote (deprecated)
- [x] katello_host_collection
- [x] katello_lifecycle_environment
- [x] katello_manifest
- [x] katello_product
- [x] katello_repository
- [x] katello_repository_set
- [x] katello_sync (the test is called repository_sync)
- [x] katello_sync_plan
- [x] katello_upload
- [x] redhat_manifest

**Comment by claudiol** [comment]
What kind of help do you need here? I'd be happy to look at this during our shutdown.

**Comment by mdellweg** [comment]
@claudiol thank you in advance.
We are looking for playbooks in `test\test_playbooks` that are self contained. They should target the features of a specific module and be able to run against an actual foreman instance. The structure would usually be something like setup -> run tests -> clean up.
For inspiration, take a look at product and organization.

**Comment by mdellweg** [comment]
Take a look at #76, too, if you like.

**Comment by claudiol** [comment]
Ok I have the realm playbook done.  I will add a pull request for you guys to review.

**Comment by akofink** [comment]
#76 is merged. #86 is ACK'd.

**Comment by mdellweg** [comment]
Since #76 is merged, the playbooks need a slightly different structure.
There are now two hosts in the inventory:
- `fixtures`: This host runs locally without modification it is meant to setup (and teardown) dependent resources, only when (re-)recording.
- `tests`: This host should run the actual tests. It is used to record the vcr-yaml-files, or to run isolated against those files.

**Comment by mdellweg** [comment]
Ping /me @sean797 @evgeni
is this list up to date?

**Comment by evgeni** [comment]
- [x] foreman_compute_profile
- [ ] foreman_compute_resource
- [x] foreman_domain
- [x] foreman_environment
- [x] foreman_global_parameter
- [x] foreman_installation_medium
- [x] foreman_job_template
- [x] foreman_location
- [x] foreman_operating_system
- [x] foreman_organization
- [x] foreman_os_default_template
- [x] foreman_provisioning_template
- [x] foreman_ptable
- [ ] foreman_realm
- [x] foreman_search_facts
- [x] foreman_setting
- [ ] foreman_setting_facts
- [x] foreman_subnet
- [x] katello_activation_key
- [x] katello_content_credential
- [x] katello_content_view
- [x] katello_content_view_filter
- [ ] katello_content_view_publish
- [ ] katello_content_view_version
- [ ] katello_content_view_version_promote
- [x] katello_lifecycle_environment
- [ ] katello_manifest
- [x] katello_product
- [x] katello_repository
- [x] katello_repository_set
- [ ] katello_sync
- [x] katello_sync_plan
- [x] katello_upload
- [x] redhat_manifest

(generated with:

```python
import glob

# copied from test_crud, no idea why I can't import that directly
TESTED_MODULES = [
    'activation_key',
    'compute_profile',
    'content_credential',
    'content_view',
    'content_view_filter',
    'domain',
    'environment',
    'global_parameter',
    'installation_medium',
    'job_template',
    'lifecycle_environment',
    'location',
    'operating_system',
    'organization',
    'os_default_template',
    'product',
    'provisioning_template',
    'ptable',
    'redhat_manifest',
    'repository',
    'repository_set',
    'repository_sync',
    'search_facts',
    'setting',
    'subnet',
    'sync_plan',
    'upload',
]

modules = glob.glob('modules/*.py')
modules.sort()

for module in modules:
    module = module.replace('modules/', '')
    module = module.replace('.py', '')
    full_module = module
    module = module.replace('foreman_', '')
    module = module.replace('katello_', '')
    if module == '__init__':
        continue
    if module in TESTED_MODULES:
        x = 'x'
    else:
        x = ' '
    print("- [{}] {}".format(x, full_module))
```

**Comment by sean797** [comment]
Awesome, we can ignore `katello_content_view_publish` & `katello_content_view_version_promote` since they are deprecated

**Comment by mdellweg** [comment]
`setting_facts` is tested together with `setting` and deprecated btw.
I think, that leaves us with `realm` and `manifest` and `compute_resource`.
For the last one, there is a promising PR https://github.com/theforeman/foreman-ansible-modules/pull/231

**Comment by mdellweg** [comment]
Getting there... Thanks @bagasse for the last addition!

**Comment by Fobhep** [comment]
@evgeni something is strange in the last overview: there is a test for compute_resource

**Comment by evgeni** [comment]
@Fobhep my tardis was broken :)

**Comment by evgeni** [comment]
updated

**Comment by evgeni** [comment]
updated @mdellweg's post with the latest list

**Comment by mdellweg** [comment]
This is to be closed with #383, right?

**Comment by evgeni** [comment]
Yes!
